### PR TITLE
Tag tweaks

### PR DESCRIPTION
**Pull request by flaurida**
* fix minor bugs and improve UX

### CODE REVIEW

File: app/components/Dashboard/Tests/index.tsx [hunk 1]
```diff
@@ -14,13 +14,17 @@ type Props = {
...
 export default function Tests({ branch, teamId }: Props): JSX.Element {
-  const { filter, tagIds } = useTagQuery();
+  const { filter, tagNames } = useTagQuery();
 
   const [search, setSearch] = useState("");
   const [checkedTestIds, setCheckedTestIds] = useState<string[]>([]);
 
   const { tests: testsData, loading } = useTests({ branch, teamId });
 
+  const testIds = useMemo(() => {
```
Comment (jperl) [inline app/components/Dashboard/Tests/index.tsx:24]: Why is this in a hook instead of just doing the map?

Comment (jperl) [reply app/components/Dashboard/Tests/index.tsx:24]: Also right below you do the map again

File: app/components/Modals/Triggers/CreateOrEditTrigger/SelectTags.tsx [hunk 2]
```diff
@@ -14,16 +14,13 @@ export default function SelectTags({
   tagIds,
   tags,
 }: Props): JSX.Element {
-  const handleClick = (tagId: string): void => {
+  const handleClick = (tagName: string): void => {
     const newTagIds = [...tagIds];
 
-    const index = newTagIds.indexOf(tagId);
+    const index = newTagIds.indexOf(tagName);
```
Comment (jperl) [inline app/components/Modals/Triggers/CreateOrEditTrigger/SelectTags.tsx:20]: is this supposed to be names?

File: app/hooks/mutations.ts [hunk 3]
```diff
@@ -655,20 +655,20 @@ export const useDeleteTag = (): MutationTuple<
...
     onCompleted: (response) => {
-      const tagId = response?.deleteTag?.id;
-      if (!tagId || !tagIds.includes(tagId)) return;
+      const tagName = response?.deleteTag?.name;
+      if (!tagName || !tagNames.includes(tagName)) return;
 
-      const newTagIds = [...tagIds];
-      const index = newTagIds.indexOf(tagId);
-      if (index > -1) newTagIds.splice(index, 1);
+      const newtagNames = [...tagNames];
```
Comment (jperl) [inline app/hooks/mutations.ts:667]: new**T**agNames

File: app/server/models/tag_trigger.ts [hunk 4]
```diff
@@ -22,16 +22,16 @@ export const updateTagTriggersForTrigger = async (
       })
       .map((tagTrigger) => tagTrigger.id);
 
-    const missingTagIds = tag_ids.filter((tag_id) => {
+    const missingTagNames = tag_ids.filter((tag_id) => {
```
Comment (jperl) [inline app/server/models/tag_trigger.ts:25]: Why are these names instead of ids?

File: app/components/Dashboard/Tests/index.tsx (hunk 1 above)
Comment (flaurida) [reply app/components/Dashboard/Tests/index.tsx:24]: it does infinite render without `useMemo`

**Pull request merged**